SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/ReferenceCountedLockingPolicy.cpp

```diff
@@ -118,24 +124,32 @@ bool UReferenceCountedLockingPolicy::IsLocked(const AActor* Actor) const
 		UE_LOG(LogReferenceCountedLockingPolicy, Warning, TEXT("IsLocked called for nullptr"));
 		return false;
 	}
-	return ActorToLockingState.Contains(Actor);
-}
 
-void UReferenceCountedLockingPolicy::OnLockedActorDeleted(AActor* DestroyedActor)
-{
-	TArray<ActorLockToken> TokensToRemove;
-	for (const auto& KeyValuePair : TokenToNameAndActor)
+	// Is this Actor explicitly locked or on a locked hierarchy ownership path.
+	if (IsExplicitlyLocked(Actor) || IsOnLockedHierarchyPath(Actor))
```

Comment:
Do we want to expose this distinction in the external API?

Reply:
I wouldn't expect us to personally, we can allow users to differentiate by having a different locking policy with the hierarchy implementation?

Reply:
I don't think it makes sense to expose this. The user shouldn't really even by querying if something is locked. If you need to rely on the fact that there is a distinction you also need to know that you are using a locking policy that has this as a implementation detail.

Reply:
We resolved to keep a single non-abstract locking policy for now - I agree with Sami on keeping IsLocked as the only implementation.